django/db/migrations/serializer.py

```diff
@@ -326,6 +362,10 @@ def serializer_factory(value):
         return RegexSerializer(value)
     if isinstance(value, uuid.UUID):
         return UUIDSerializer(value)
+    if hasattr(value, "__module__") and hasattr(value, "__class__"):
+        class_serializer = ClassSerializer(value)
+        if class_serializer.serialize():
+            return class_serializer
```

Comment:
The fact that you have to call `serialize` to determine if the serializer is appropriate differs from the pattern used with the other ones. To me that's a code smell that hints at the fact we're abusing the serializer pattern here.